lib/marionette/drivers/tcp-sync.js

```diff
@@ -60,6 +60,10 @@ TcpSync.prototype.waitForSocket = function(options, callback) {
   var start = Date.now();
   var self = this;
 
+  // Use sockittome's built in polling timeout during calls to connect
+  // to avoid socket misuse.
+  sockit.setPollTimeout(interval);
```

Comment:
Nah, I think having about 10 tries in 10 seconds will be good for us. If we add that extra it will mean less attempts.